### ISSUE TITLE
Running INSERT statements against MySQL database results in Exception but value is inserted

### ISSUE THREAD

**Issue by tobias-gruenewald** [mithrandyr/SimplySql#3]
Invoke-SqlQuery -ConnectionName ConName -Query "INSERT INTO table (column) VALUES ('value');"

Invoke-SqlQuery : Die Tabelle 0 kann nicht gefunden werden.
At line:1 char:5
+     Invoke-SqlQuery -ConnectionName Postfix -Query "INSERT INTO table ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [Invoke-SqlQuery], IndexOutOfRangeException
    + FullyQualifiedErrorId : System.IndexOutOfRangeException,Invoke-SqlQuery

This happens for any database, table and column I tried with. The value is inserted correctly in the table but the Cmdlet still throws an exception. SELECT or DELETE statements work without any issues.

Please let me know what kind of details would help to identify the root cause.

**Comment by mithrandyr** [comment]
@tobias-gruenewald I was able to reproduce your issue and I have identified what is happening.
```powershell
Open-SQLiteConnection
Invoke-SqlUpdate -Query "CREATE TABLE temp (cola int, colb text)"
Invoke-SqlQuery -Query "INSERT INTO temp VALUES (1, 'alpha')" # this will generate an error but insert
Invoke-SqlUpdate -Query "INSERT INTO temp VALUES (2, 'beta')" #this will NOT generate an error
Invoke-SqlQuery -Query "SELECT * FROM temp" # showing both rows
Close-SqlConnection
````
The Invoke-SqlQuery cmdlets uses a datatableadapter in order to return data -- when you don't pass in a SQL statement that returns a resultset, then an error is thrown.  To avoid this, use the Invoke-SqlUpdate cmdlets for all data modification queries.

SimplySql has three separate cmdlets to handle executing queries.  Invoke-SqlQuery is for queries that return results, Invoke-SqlScalar is a specialized version of Invoke-SqlQuery that returns the first column of the first row of the first table in the resultset (and returns it as a primitive rather than an object or datarow), Invoke-SqlUpdate is for all queries that don't return results.

Invoke-SqlQuery uses .NET dataTableAdapaters or .NET DataReader (when used w/ the -Stream option).
Invoke-SqlScalar uses .NET dataCommand.ExecuteScalar() function.
Invoke-SqlUpdate uses .NET dataCommand.ExecuteNonQuery() function.

I'll make an update to the Invoke-SqlQuery cmdlet to throw a warning instead of an error when no results are returned.

**Comment by tobias-gruenewald** [comment]
Thanks for the very fast feedback.
Will use Invoke-SqlUpdate in my code now, I was not aware of that Cmdlet.

Keep up the good work. It's a very nice Powershell Tool that helps me out a great deal!

**Comment by mithrandyr** [comment]
@tobias-gruenewald No problem, will be releasing a minor update in the next few minutes (1.3.8) that will change this from an error to a warning with the following text:

> Query returned no resultsets.  This occurs when the query has no select statement or invokes a stored procedure that doesn't return a resultset.  Use 'Invoke-SqlUpdate' to avoid this warning.

Also, this project does have a wiki that walks you through each command [SimplySql Wiki](https://github.com/mithrandyr/SimplySql/wiki) -- let me know if there is anything that I can add that will improve the documentation.  Thanks for using the module and glad it helps you out!

**Comment by mithrandyr** [comment]
Fixed in v1.3.8 -- you can pull from powershellgallery.com with update-module simplysql
